lib/shopify_transporter/exporters/magento/order_exporter.rb

```diff
@@ -12,14 +12,21 @@ def initialize(store_id: nil, soap_client: nil, database_adapter: nil)
 
         def export
           $stderr.puts 'Starting export...'
-          base_orders.map do |order|
-            $stderr.puts "Fetching order: #{order[:increment_id]}..."
-            order.merge(items: info_for(order[:increment_id]))
+          puts '['
```

Comment:
I think we should add tests that check the output works with multiple objects and not just one.

You mentioned this, but I'm hugely in favor of moving some of this code up out of each of the `export` methods.

What if we add a `print_exported_objects` method in `lib/shopify_transporter/exporters/exporter.rb` and made these export methods yield objects instead?

```
def run
  print_exported_objects
end

def print_exported_objects
  puts '['
  first_object = true
  Magento::MagentoExporter
    .for(type: object_type)
    .new(
      store_id: config['export_configuration']['store_id'],
      soap_client: soap_client,
      database_adapter: database_adapter
    )
    .export_objects do |object|
      if first_object
        puts ','
        first_object = false
      end
      print '  ' + JSON.pretty_generate(object, object_nl: "\n  ")
    end
  ensure
  puts ']'
end


# In exporters - we can probably use a better name.
def export_objects
  $stderr.puts 'Starting export...'
  base_orders.each do |base_order|
    puts ',' unless base_order == base_orders.first
    $stderr.puts "Fetching order: #{base_order[:increment_id]}..."
    yield with_attributes(base_order)
  end
end
```

Then all the JSON code can be tested in one place.

Reply:
I like this idea. I'd go as far as to extract the logger to its own class. It's pretty standalone.